ivy/functional/backends/numpy/experimental/statistical.py

```diff
@@ -41,3 +42,28 @@ def unravel_index(
     out: Optional[np.ndarray] = None,
 ) -> np.ndarray:
     return np.unravel_index(indices, shape)
+
+
+def quantile(
+    a: np.ndarray,
+    q: Union[float, np.ndarray],
+    /,
+    *,
+    axis: Optional[Union[int, Sequence[int]]] = None,
+    keepdims: bool = False,
+    interpolation: str = 'linear',
+    out : Optional[np.ndarray] = None 
+) -> np.ndarray:
+    # quantile method in numpy backend, always return an array with dtype=float64.
+    # in other backends, the output is the same dtype as the input.
+    input_dtype = str(a.dtype)
+
+    tuple(axis) if isinstance(axis, list) else axis
+    return np.quantile(
+        a,
+        q,
+        axis=axis,
+        method=interpolation,
+        keepdims=keepdims,
+        out=out
+    ).astype(input_dtype)
```

Comment:
Could this have been `.astype(a.dtype)` perhaps?

Reply:
yes, I changed it and it passed the tests. At the beginning I was worried about different assignment of float type, I think I had some problem with Torch that would not accept float16 as an input, and it has to be float32 or above. I was trying to assign right data type instead of converting to float using Numpy instance method.